src/doc/rustdoc/src/linking-to-items-by-name.md

```diff
@@ -24,11 +24,29 @@ pub struct Foo4;
 pub struct Bar;
 ```
 
+Unlike normal markdown, `[bar][Bar]` syntax is also supported without needing a
+`[Bar]: ...` reference link, and links are case-sensitive.
```

Comment:
We should probably note somewhere that non-intra-doc links are case-**insensitive**; see #80882.